### ISSUE TITLE
causal conv eager issue

### ISSUE THREAD

**Issue by flowpoint** [keras-team/keras#11349]
### eager mode doesnt work with causal conv:
` Invalid padding method 'CAUSAL' ` 
gets thrown by [this code](https://gist.github.com/flowpoint/ae2d665b0606694dcb635b5074d078ed) :
```
import tensorflow as tf
#import keras # code works without eager
#but with it it wants tf.contrib.eager.Variable somewhere

from tensorflow import keras # <- errors on padding='causal'

import numpy as np
#tf.enable_eager_execution()

data = np.random.random([1,1,256])

leaky_relu = tf.nn.leaky_relu
model = keras.Sequential([
    keras.layers.Conv1D(16, 2, padding='causal', dilation_rate=2, activation=leaky_relu) ])

opt = tf.train.AdamOptimizer()
model.compile(opt, loss = 'mean_squared_error', metrics=['accuracy'])

model.fit(data, y=np.random.random([1,1,16]), batch_size=1) 
```

tensorflow==1.12.0rc0
keras==2.2.4

-> collision between keras and tensorflow eagerness

**Comment by gabrieldemarmiesse** [comment]
Keras doesn't support tensorflow eager yet. Likely in TF 2.0. You should use tf.keras. If you have a problem with tf.keras, you need to open an issue in the tensorflow repository.
